src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -333,97 +244,617 @@ struct jit_uni_mvn_mean_variance_kernel_f32 : public jit_uni_mvn_mean_variance_k
+                        }
+                        uni_vsubps(Vmm(5), Vmm(5), ptr[reg_mean_active]);
+                        uni_vfmadd231ps(Vmm(9), Vmm(5), Vmm(5));
+                        add(reg_mean_active, 4 * sizeof(float));
+                    } else {
+                        if (!isFloatCompatible(jcp_.src_prc))
+                            uni_vpaddd(Vmm(9), Vmm(9), Vmm(5));
+                        else
+                            uni_vaddps(Vmm(9), Vmm(9), Vmm(5));
+                    }
+                    break;
+                case 2:
+                    load_tail2_emitter->emit_code({static_cast<size_t>(reg_src_aux.getIdx())}, {static_cast<size_t>(6)},
+                                               {}, {load_pool_gpr_idxs});
+                    if (jcp_.normalize_variance) {
+                        if (!isFloatCompatible(jcp_.src_prc)) {
+                            uni_vcvtdq2ps(Vmm(6), Vmm(6));
+                        }
+                        uni_vsubps(Vmm(6), Vmm(6), ptr[reg_mean_active]);
+                        uni_vfmadd231ps(Vmm(10), Vmm(6), Vmm(6));
+                        add(reg_mean_active, 2 * sizeof(float));
+                    } else {
+                        if (!isFloatCompatible(jcp_.src_prc))
+                            uni_vpaddd(Vmm(10), Vmm(10), Vmm(6));
+                        else
+                            uni_vaddps(Vmm(10), Vmm(10), Vmm(6));
+                    }
+                    break;
+                case 1:
+                    load_tail1_emitter->emit_code({static_cast<size_t>(reg_src_aux.getIdx())}, {static_cast<size_t>(7)},
+                                               {}, {load_pool_gpr_idxs});
+                    if (jcp_.normalize_variance) {
+                        if (!isFloatCompatible(jcp_.src_prc)) {
+                            uni_vcvtdq2ps(Vmm(7), Vmm(7));
+                        }
+                        uni_vsubps(Vmm(7), Vmm(7), ptr[reg_mean_active]);
+                        uni_vfmadd231ps(Vmm(11), Vmm(7), Vmm(7));
+                        add(reg_mean_active, 1 * sizeof(float));
+                    } else {
+                        if (!isFloatCompatible(jcp_.src_prc))
+                            uni_vpaddd(Vmm(11), Vmm(11), Vmm(7));
+                        else
+                            uni_vaddps(Vmm(11), Vmm(11), Vmm(7));
+                    }
+                    break;
+                default:
+                    assert(!"MVN layer tails is processed only with 8/4/2/1 blocks.");
+                    break;
+                }
+            };
+
+            Label tail_blk8_label;
+            Label tail_blk8_exit_label;
+            Label tail_blk4_label;
+            Label tail_blk4_exit_label;
+            Label tail_blk2_label;
+            Label tail_blk2_exit_label;
+            Label tail_blk1_label;
+            Label tail_blk1_exit_label;
+            L(tail_blk8_label);
```

Comment:
General ask for the whole PR: need to get rid of code duplication. Most of it produced due to necessety to handle tiles with different granularity. Possible solution is to create loop over array with this tail size {8, 4, 2, 1} and generate code for its tile size by loop body in unified way.

Reply:
Refactored in commit:https://github.com/openvinotoolkit/openvino/pull/17988/commits/621014e23bb2096fe8cc7aa1238698779223f4e4
Due to performance consideration, 8+4+2+1 tails should be processed continues, otherwise loop for each tile will go through the whole tensor, as this is nspc layout. I extract the process logic of tiles and vectors, accepting different worker type. This could help clean tails part as well as vector part. Thanks Dmitry!

Reply:
By loops I mentioned code generation level loops, not execution time. See my comment with example: https://github.com/openvinotoolkit/openvino/pull/17988#discussion_r1262211384
What do you think?

Reply:
I see, with loop of tile number, we have clean code, and no more compilation time and compiled kernel size increased. Thanks a lot Dmitry! done in [d464ca6](https://github.com/openvinotoolkit/openvino/pull/17988/commits/d464ca6a2cb92f858670f01553892b975c0c20f5)